### PR TITLE
Adding basic monitor endpoint

### PR DESCRIPTION
**Pull request by chouseknecht**
Based on PR #2172   
Applies changes requested by @cutwater. 

Adds `/api/monitor` endpoint that returns a basic `ok` or `error` state for PostgreSQL, RabbitMQ, Redis, and InfluxDB.

### CODE REVIEW

File: galaxy/api/views/monitor.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        targets = executor.loader.graph.leaf_nodes()
+
+        if executor.migration_plan(targets):
+            migrations = 'needed'
+    except DatabaseError:
+        status = 'error'
+    return status, migrations
+
+
+def get_redis_status():
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]:
```

def get_redis_status():
    client = redis.Redis(
        host=settings.REDIS_HOST,
        port=settings.REDIS_PORT
    )
    try:
        redis_client.ping()
    except redis.RedisError:
        return 'error'
    except Exception:
        logger.exception('Unexpected error when trying to connect to Redis')
        return 'error'
    finally:
        redis.close()
    return 'ok'

```

File: galaxy/api/views/monitor.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+import logging
+from collections import OrderedDict
+import socket
+
+from django.conf import settings
+from django.db.migrations.executor import MigrationExecutor
+from django.db import connections, DatabaseError, DEFAULT_DB_ALIAS
+import influxdb
+from kombu import Connection as RabbitMQ
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]: `import kombu`

Also using an alias for `kombu.Connection` is misleading. `kombu.Connection` is not limited to RabbitMQ only.

File: galaxy/api/views/monitor.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        port=settings.REDIS_PORT
+    )
+    try:
+        redis_client.ping()
+    except redis.exceptions.RedisError:
+        response = 'error'
+    return response
+
+
+def get_rabbit_status():
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]:
```
def get_rabbit_status():
    client = kombu.Connection(settings.BROKER_URL)
    with client:
        try:
            client.connect()
        except client.connection_errors:
            return 'error'
        except Exception:
            logger.exception(
                'Unexpected error when trying to connect to RabbitMQ'
            )
            return 'error'
    return 'ok'
```

File: galaxy/api/views/monitor.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+    """ Monitor resources """
+    permission_classes = (AllowAny,)
+
+    def get(self, request, format=None):
+        response = OrderedDict()
+        response['influx'] = get_influx_status()
+        (response['postgresql'], response['migrations']) = \
+            get_postgres_status()
+        response['redis_status'] = get_redis_status()
+        response['rabit'] = get_rabbit_status()
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]: typo: rabbit

File: galaxy/api/views/monitor.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+class MonitorRootView(base_views.APIView):
+    """ Monitor resources """
+    permission_classes = (AllowAny,)
+
+    def get(self, request, format=None):
+        response = OrderedDict()
+        response['influx'] = get_influx_status()
+        (response['postgresql'], response['migrations']) = \
+            get_postgres_status()
+        response['redis_status'] = get_redis_status()
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]: please use consistent naming, choose one: with or without `_status` suffix.

File: galaxy/api/views/monitor.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+    rabbit_client.close()
+    return response
+
+
+class MonitorRootView(base_views.APIView):
+    """ Monitor resources """
+    permission_classes = (AllowAny,)
+
+    def get(self, request, format=None):
+        response = OrderedDict()
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]: minor: OrderedDict() may not be needed.

**Comment by till** [comment]
@chouseknecht I am happy to finish this off if you are too busy this week. I'd really like to push this over the line. LMK

File: galaxy/api/views/monitor.py [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+    )
+    try:
+        client.ping()
+    except redis.RedisError:
+        return 'error'
+    except Exception:
+        logger.exception('Unexpected error when trying to connect to Redis')
+        return 'error'
+    finally:
+        del client
```
Comment (cutwater) [inline galaxy/api/views/monitor.py]: Using `del` operator does not guarantee deletion of the object and releasing of resources.

**Pull request merged**